pkg/operator/ceph/object/admin.go

```diff
@@ -80,12 +80,25 @@ func NewMultisiteContext(context *clusterd.Context, clusterInfo *cephclient.Clus
 func extractJSON(output string) (string, error) {
 	// `radosgw-admin` sometimes leaves logs to stderr even if it succeeds.
 	// So we should skip them if parsing output as json.
-	pattern := regexp.MustCompile(`(?ms)^{.*}$`)
-	match := pattern.Find([]byte(output))
-	if match == nil {
+	// valid JSON can be an object (in braces) or an array (in brackets)
+	arrayRegex := regexp.MustCompile(`(?ms)^\[.*\]$`)
+	arrayMatch := arrayRegex.Find([]byte(output))
+	objRegex := regexp.MustCompile(`(?ms)^{.*}$`)
+	objMatch := objRegex.Find([]byte(output))
+	if arrayMatch == nil && objMatch == nil {
 		return "", errors.Errorf("didn't contain json. %s", output)
 	}
-	return string(match), nil
+	if arrayMatch == nil && objMatch != nil {
+		return string(objMatch), nil
+	}
+	if arrayMatch != nil && objMatch == nil {
+		return string(arrayMatch), nil
+	}
+	// if both object and array match, take the largest of the two matches
+	if len(arrayMatch) > len(objMatch) {
+		return string(arrayMatch), nil
+	}
+	return string(objMatch), nil
 }
```

Comment:
@satoru-takeuchi added this recently. The unit tests should prove it out pretty well, but let me know if you have any concerns.

Reply:
@BlaineEXE I'll anser because today is my holiday.

Reply:
@BlaineEXE I guess `radosgw-admin` returns either array or object. Have you ever seens the output which consists of both?

In addition, the meaning of L98's if statement is unclear. I prefer to regard this case as an error. If `radosgw-admin` may returns both, it's better to return both `string(arrayMatch)` and `string(objMatch)`, and caller decides how to use them.

Reply:
Well, when it returns an array, the array is an array of objects. I haven't seen where it returns an object that contains arrays, but it isn't hard to imagine that being a possibility.

Reply:
Ah, sorry, my description was ambiguous. What I meant was outputs having both array starting at the beginning of line, and such objects. They matches matches "(?ms)^{.*}$"  
 or "(?ms)^{.*}$`". 

Since output of `radosgw-admin` is formatted, the output is either

```
{
   ...
}
```

or

```
[
  ...
]
```
Hence arrays in object and objects in arrays start after one or more spaces/tabs and don't match the above-mentioned regexps.

Reply:
Ah. I see. I think it's still good to keep the logic of picking the largest match just in case or in case we want to reuse this for something else later.